omero_mapr/show.py

```diff
@@ -163,35 +174,66 @@ def mapr_paths_to_object(conn, mapann_ns=[], mapann_query=None,
         q_select.append(" sl.parent.id as screen_id ")
     elif plate_id:
         q_select.append(" sl.parent.id as screen_id ")
-        q_select.append(" p.id as plate_id")
+        q_select.append(" pl.id as plate_id")
+    elif project_id:
+        q_select.append(" pdl.parent.id as project_id ")
+    elif dataset_id:
+        q_select.append(" pdl.parent.id as project_id ")
+        q_select.append(" ds.id as dataset_id")
     elif image_id:
-        q_select.append(" sl.parent.id as screen_id ")
-        q_select.append(" p.id as plate_id ")
+        q_select.append(" COALESCE(sl.parent.id,null) as screen_id ")
+        q_select.append(" COALESCE(pdl.parent.id,null) as project_id ")
+        q_select.append(" COALESCE(pl.id,null) as plate_id ")
+        q_select.append(" COALESCE(ds.id,null) as dataset_id ")
         q_select.append(" i.id as image_id ")
 
     if q_select:
         q += ", ".join(q_select) + ", "
 
     q += ''' count(i.id) as imgCount)
```

Comment:
why not following the same strategy for ``q_select`` and ``q`` i.e. use ``append`` instead of ``+``